web3/middleware/cache.py

```diff
@@ -92,20 +89,25 @@ def simple_cache_middleware(
     ) -> Callable[[RPCEndpoint, Any], RPCResponse]:
         lock = threading.Lock()
 
+        # Setting the cache here, rather than in ``construct_simple_cache_middleware``,
```

Comment:
Might consider adding a couple words for users about the ability to use multiple web3/cache instances in the ``construct_simple_cache_middleware`` comment.

Reply:
I'm always for more clarification. I'm not quite sure what to put here though. Do you mean in the docstring description? Imo, this is fixing a broken case since middleware should never be expected to share state. Maybe the general middleware docs could use some updating with a sentence or two to reflect multiple instance support? Thoughts?

Reply:
> I'm always for more clarification. I'm not quite sure what to put here though. Do you mean in the docstring description? Imo, this is fixing a broken case since middleware should never be expected to share state. Maybe the general [middleware docs](https://web3py.readthedocs.io/en/latest/middleware.html#middleware) could use some updating with a sentence or two to reflect multiple instance support? Thoughts?

I agree with @fselmo here. I opened the original issue as I understood that multiple providers would be supported by the cache middleware and I was surprised to find out that state was shared. Also, current behavior seems to be a regression from previous web3py versions, so not sure why docs should be updated now.

Reply:
@reedsa @pacrob I took this time to update the docs (which were pretty outdated). But lmk if you meant something else. If so, maybe it's beyond the scope of this PR.

Reply:
Thank you, the docs are a great place to outline this.